app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -175,40 +175,44 @@ object LayerCache extends Config with LazyLogging with KamonTrace {
     subNode: Option[UUID],
     user: User,
     voidCache: Boolean = false
-  ): OptionT[Future, (MapAlgebraAST, EvalParams)] = {
-    val cacheKey = s"ast-params-$toolRunId-${subNode}-${user.id}"
-    if (voidCache) rfCache.delete(cacheKey)
-    rfCache.cachingOptionT(cacheKey, doCache = cacheConfig.tool.enabled) {
-      for {
-        (tool, toolRun) <- toolAndToolRun(toolRunId, user)
-        oldAst   <- OptionT.fromOption[Future]({
-          logger.debug(s"Parsing Tool AST with ${tool.definition}")
-          val entireAST = tool.definition.as[MapAlgebraAST].valueOr(throw _)
-          subNode.flatMap(id => entireAST.find(id)).orElse(Some(entireAST))
-        })
-        subs     <- assembleSubstitutions(oldAst, { id: UUID =>
-          OptionT(Tools.getTool(id, user))
-            .map({ referrent => referrent.definition.as[MapAlgebraAST].valueOr(throw _) })
-            .value
-        })
-        ast      <- OptionT.fromOption[Future](oldAst.substitute(subs))
-        nodeId   <- OptionT.pure[Future, UUID](subNode.getOrElse(ast.id))
-        params   <- OptionT.pure[Future, EvalParams]({
-          logger.debug(s"Parsing ToolRun parameters with ${toolRun.executionParameters}")
-          val parsedParams = toolRun.executionParameters.as[EvalParams].valueOr(throw _)
-          val defaults = ast.metadata
-          val overrides = parsedParams.metadata.get(ast.id)
-          val md = (overrides |@| defaults).map(_.fallbackTo(_))
-            .orElse(overrides)
-            .orElse(defaults)
-            .getOrElse(NodeMetadata())
-
-          EvalParams(
-            parsedParams.sources,
-            parsedParams.metadata + (ast.id -> md)
-          )
-        })
-      } yield (ast, params)
+  ): OptionT[Future, (MapAlgebraAST, EvalParams)] =
+    traceName("LayerCache.toolEvalRequirements") {
+      val cacheKey = s"ast+params-$toolRunId-${subNode}-${user.id}"
+      if (voidCache) histogramCache.delete(cacheKey)
+      astCache.cachingOptionT(cacheKey) { implicit ec =>
```

Comment:
I'm getting compiler errors on this line and the one above, `astCache` and `histogramCache` are not found.